linkerd/proxy/tap/src/accept.rs

```diff
@@ -70,15 +70,19 @@ impl Service<Connection> for AcceptPermittedClients {
         match meta.peer_identity {
             Conditional::Some(ref peer) => {
                 if self.permitted_client_ids.contains(peer) {
-                    self.serve_authenticated(io)
+                    future::ok(self.serve_authenticated(io))
                 } else {
-                    self.serve_unauthenticated(io, format!("Unauthorized peer: {}", peer))
+                    future::ok(
+                        self.serve_unauthenticated(io, format!("Unauthorized peer: {}", peer)),
+                    )
                 }
             }
             Conditional::None(ReasonForNoIdentity::NoPeerName(ReasonForNoPeerName::Loopback)) => {
-                self.serve_authenticated(io)
+                future::ok(self.serve_authenticated(io))
+            }
+            Conditional::None(reason) => {
+                future::ok(self.serve_unauthenticated(io, reason.to_string()))
```

Comment:
Nit, take it or leave it: it looks like we wrap each match arm in `future::ok`. We should be able to simplify this a little by wrapping the entire match in a `future::ok`, like:
```rust
future::ok(match meta.peer_identity { 
    Conditional::Some(ref peer) => {
        if self.permitted_client_ids.contains(peer) {
            self.serve_authenticated(io)
        } else {
            // ...
        }
    }
    // ...
})
```